### ISSUE TITLE
Tab not working in parse_key_value

### ISSUE THREAD

**Issue by Christoph-Karl** [vectordotdev/vector#14630]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Problem

Using "parse_key_value" to parse a log file which contains a tab ("\t") does not work.
It produces the following output:

```text
2022-09-29T08:30:20.836410Z ERROR transform{component_kind="transform" component_id=modify component_type=remap component_name=modify}: vector::internal_events::remap: Mapping failed with event. error="function call error for \"parse_key_value\" at (10:86): could not parse whole line successfully" error_type="conversion_failed" stage="processing" internal_log_rate_secs=10
{"file":"logs_err","message":"a:b\tx:y","source_type":"file","timestamp":"2022-09-29T08:30:20.835910156Z"}
```

Expected behaviour:

```text
{"file":"logs_ok","message":"a:b\tx:y","source_type":"file","timestamp":"2022-09-29T08:36:40.174592882Z","values":{"a":"b","x":"y"}}
```

### Configuration

```text
.values = parse_key_value!(.message, key_value_delimiter: ":", field_delimiter: "\t",)
```

### Version

vector 0.24.1 (x86_64-unknown-linux-musl 8935681 2022-09-12)

### Example Data

a:b      x:y
There should be a tab ("\t") between "b" and "x"!

### Additional Context

Same example works if I use a "_" instead of tab.

**Comment by Christoph-Karl** [comment]
Does still not work with vector 0.25.0

**Comment by jszwedko** [comment]
@Christoph-Karl this will be included in 0.26.0 since it was merged after the release branch was cut.
